keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -324,49 +340,71 @@ def next(prompt, state, index):
 
     def generate(
         self,
-        prompt,
-        max_length,
+        inputs,
+        max_length=None,
+        batch_size=None,
     ):
         """Generate text.
 
-        This method generates text based on given `prompt`. Generation will
-        continue until `max_length` is met, and all tokens generated after
-        `end_token` will be truncated. The sampling approach used can be
-        controlled via the sampler argument.
+        This method generates text based given `inputs` as a prompt.
 
         Args:
-            prompt: a string, string Tensor or string RaggedTensor. The prompt
-                text for generation.
+            inputs: a list of strings, `tf.Tensor`, or `tf.data.Dataset`.
```

Comment:
Starting with "a list" is a bit confusing (wrt Tensor and Dataset case)

"a string `tf.Tensor`, a `tf.data.Dataset` of strings, or a list of Python strings"